### PR TITLE
[AAE-4428] Add selected file counter to attach file widget

### PR DESCRIPTION
**Pull request by TomikaArome**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

No way to tell exactly how many files are selected without counting them manually.

**What is the new behaviour?**

A counter in the toolbar to the right of the breadcrumb with the number of files selected is displayed.
image

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — CHANGES_REQUESTED** [review]
missing unit test

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts [hunk 1]
```diff
@@ -278,6 +280,14 @@ export class ContentNodeSelectorPanelComponent implements OnInit, OnDestroy {
         return this._chosenNode;
     }
 
+    getSelectedCountTranslation(): string {
+        return this.translation.instant(`NODE_SELECTOR.SELECTED_COUNT`, { count: this.getSelectedCount() });
```
Comment (arditdomi) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts:284]: you could do the same in the html and let the translate pipe do the job by doing:

`'NODE_SELECTOR.SELECTED_COUNT' | translate: { count: getSelectedCount() }`

Not much different from what you are doing though, just less number of services injected in the constructor in this way as you don't need to inject the translation service, but up to you to decide which one you want to use

Comment (TomikaArome) [reply lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts:284]: Good idea, I will change it

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.html [hunk 2]
```diff
@@ -57,6 +57,10 @@
                 [root]="breadcrumbFolderTitle"
                 data-automation-id="content-node-selector-content-breadcrumb">
             </adf-dropdown-breadcrumb>
+            <div
+                data-automation-id="content-node-selector-selected-counter">
```
Comment (arditdomi) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.html:61]: this data-automation-id is not used anywhere

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.spec.ts [hunk 3]
```diff
@@ -1304,5 +1304,24 @@ describe('ContentNodeSelectorPanelComponent', () => {
                 expect(component.sorting).toEqual(['createdAt', 'desc']);
             });
         });
+
+        describe('Selected nodes counter', () => {
```
Comment (arditdomi) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.spec.ts:1308]: Could you add an extra check that checks the binding with the html? Right now your test is checking the getselectedCount() method but if someone changes the html that you added, the counter could be broken without any tests highlighting a problem

**Review by arditdomi — CHANGES_REQUESTED** [review]

File: lib/content-services/src/lib/content-node-selector/content-node-selector.component.spec.ts [hunk 4]
```diff
@@ -393,4 +394,28 @@ describe('ContentNodeSelectorComponent', () => {
...
+            component.onSelect([new Node({ id: 'fake' })]);
+
+            expect(component.getSelectedCount()).toBe(1);
+        });
+
+        it('should have injected the counter text', () => {
+            fixture.detectChanges();
+            fixture.whenStable().then(() => {
+                const breadcrumb = fixture.debugElement.query(By.directive(DropdownBreadcrumbComponent));
+                expect(breadcrumb).not.toBeNull('Breadcrumb not found');
```
Comment (arditdomi) [inline lib/content-services/src/lib/content-node-selector/content-node-selector.component.spec.ts:413]: Why do we need to check the breadcrumb?

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.html (hunk 2 above)
Comment (mauriziovitale) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.html:61]: is the counter visible on the local upload tab too? 
I was expecting to have a directive

File: lib/content-services/src/lib/content-node-selector/content-node-selector.component.spec.ts (hunk 4 above)
Comment (TomikaArome) [reply lib/content-services/src/lib/content-node-selector/content-node-selector.component.spec.ts:413]: Not sure why I left that in. I think I assumed the breadcrumb would have to be present for the counter to also be present, when that's not true. Removing.

**Review by arditdomi — APPROVED** [review]

**Review by mauriziovitale — APPROVED** [review]

File: lib/content-services/src/lib/directives/node-counter.directive.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    componentRef: NodeCounterComponent;
+
+    constructor(
+        private resolver: ComponentFactoryResolver,
+        public viewContainerRef: ViewContainerRef
+    ) {}
+
+    ngOnInit() {
+        const componentFactory = this.resolver.resolveComponentFactory(NodeCounterComponent);
```
Comment (DenysVuika) [inline lib/content-services/src/lib/directives/node-counter.directive.ts:35]: why do you need dynamically creating same component?

Comment (arditdomi) [reply lib/content-services/src/lib/directives/node-counter.directive.ts:35]: It is inside the directive that we create the NodeCounterComponent. Not sure if I answered your question or not 😅

**Review by DenysVuika — APPROVED** [review]

**Review by eromano — CHANGES_REQUESTED** [review]
missing documentation of the new directive

**Comment by TomikaArome** [comment]
> missing documentation of the new directive

Documentation added. Do we need to wait for the build again? It was green before.

**Review by eromano — APPROVED** [review]

**Pull request merged**